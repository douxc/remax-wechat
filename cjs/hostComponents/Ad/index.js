"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Ad = void 0;
var runtime_1 = require("@remax/runtime");
exports.Ad = (0, runtime_1.createHostComponent)('ad');
exports.Ad.defaultProps = {
    adType: 'banner',
    adTheme: 'white',
};
