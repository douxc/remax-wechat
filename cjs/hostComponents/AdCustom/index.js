"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdCustom = void 0;
var runtime_1 = require("@remax/runtime");
/**
 * 原生模版广告
 * @see https://developers.weixin.qq.com/miniprogram/dev/component/ad-custom.html
 */
exports.AdCustom = (0, runtime_1.createHostComponent)('ad-custom');
