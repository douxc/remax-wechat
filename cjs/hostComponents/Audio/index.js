"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Audio = void 0;
var runtime_1 = require("@remax/runtime");
// 微信已不再维护
exports.Audio = (0, runtime_1.createHostComponent)('audio');
