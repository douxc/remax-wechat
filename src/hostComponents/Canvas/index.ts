import * as React from 'react';
import { createHostComponent } from '@remax/runtime';
import { BaseProps, GenericEvent } from '../../types/component';

export interface CanvasProps extends BaseProps {
  /** 指定 canvas 类型，当前仅支持 webgl 2.7.0 */
  type?: string;
  /** canvas 组件的唯一标识符，若指定了 type 则无需再指定该属性 1.0.0 */
  canvasId?: string;
  /** 当在 canvas 中移动时且有绑定手势事件时，禁止屏幕滚动以及下拉刷新 1.0.0 */
  disableScroll?: boolean;
  /** 当发生错误时触发 error 事件，detail = {errMsg} 1.0.0  */
  onError?: (event: GenericEvent) => any;
}

/**
 * 画布。2.9.0 起支持一套新 Canvas 2D 接口（需指定 type 属性），同时支持同层渲染，原有接口不再维护。旧版本可参考 旧版画布迁移指南 进行迁移。
 * @see https://developers.weixin.qq.com/miniprogram/dev/component/canvas.html
 */
export const Canvas: React.ComponentType<CanvasProps> =
  createHostComponent<CanvasProps>('canvas');

Canvas.defaultProps = {
  disableScroll: false,
};
