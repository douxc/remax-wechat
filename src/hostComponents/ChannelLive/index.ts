import * as React from 'react';
import { createHostComponent } from '@remax/runtime';
import { BaseProps } from '../../types/component';

export interface ChannelLiveProps extends BaseProps {
  /** 视频 feedId */
  feedId: string;
  /** 视频号 id，以“sph”开头的id，可在视频号助手获取。视频号必须与当前小程序相同主体。 */
  finderUserName: string;
}
/**
 * 基础库 2.29.0 开始支持
 * https://developers.weixin.qq.com/miniprogram/dev/component/channel-live.html
 */
export const ChannelLive: React.ComponentType<ChannelLiveProps> =
  createHostComponent<ChannelLiveProps>('channel-live');
