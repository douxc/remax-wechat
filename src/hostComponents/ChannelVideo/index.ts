import * as React from 'react';
import { createHostComponent } from '@remax/runtime';
import { BaseProps, GenericEvent } from '../../types/component';

export interface ChannelVideoProps extends BaseProps {
  /** 仅视频号视频与小程序同主体时生效。若内嵌非同主体视频，请使用 feed-token。 */
  feedId: string;
  /** 视频号 id，以“sph”开头的id，可在视频号助手获取。 */
  finderUserName: string;
  /** 仅内嵌小程序非同主体视频号视频时使用，获取方式参考本指引。 */
  feedToken: string;
  /** 是否自动播放。仅视频号视频与小程序同主体时支持设置为 true。 */
  autoplay: string;
  /** 是否循环播放 */
  loop?: boolean;
  /** 是否静音播放 */
  muted?: boolean;
  /** 当视频大小与 video 容器大小不一致时，视频的表现形式 */
  objectFit?: boolean | 'contain' | 'cover' | 'fill';
  onError?: (e: GenericEvent) => void;
}

/**
 * 基础库 2.25.1 开始支持
 * 小程序内嵌视频号视频组件，支持在小程序中播放视频号视频，并无弹窗跳转至视频号
 * https://developers.weixin.qq.com/miniprogram/dev/component/channel-video.html
 */
export const ChannelVideo: React.ComponentType<ChannelVideoProps> =
  createHostComponent<ChannelVideoProps>('channel-video');

ChannelVideo.defaultProps = {
  loop: false,
  muted: false,
  objectFit: 'contain',
};
