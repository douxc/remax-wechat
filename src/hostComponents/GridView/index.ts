import * as React from 'react';

import { createHostComponent } from '@remax/runtime';
import { BaseProps } from '../../types/component';

interface GridViewProps extends BaseProps {
  /**
   * 布局方式
   * aligned - 每行高度由同一行中最大高度子节点决定
   * masonry - 瀑布流，根据子元素高度自动布局
   *
   * default: aligned
   */
  type: 'aligned' | 'masonry';
  /**
   * 交叉轴元素数量
   *
   * default: 2
   */
  crossAxisCount?: number;
  /**
   * 交叉轴元素最大范围
   *
   * default: 0
   */
  maxCrossAxisExtent?: number;
  /**
   * 主轴方向间隔
   *
   * default: 0
   */
  mainAxisGap?: number;
  /**
   * 交叉轴方向间隔
   *
   * default: 0
   */
  crossAxisGap?: number;
}

export const GridView: React.ComponentType<GridViewProps> =
  createHostComponent<GridViewProps>('grid-view');
