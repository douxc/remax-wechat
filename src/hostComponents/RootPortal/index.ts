import * as React from 'react';
import { createHostComponent } from '@remax/runtime';
import { BaseProps } from '../../types/component';

interface RootPortalProps extends BaseProps {
  /**
   * 是否从页面中脱离出来
   *
   * 2.26.1
   *
   * default: true
   */
  enable?: boolean;
}

export const RootPortal: React.ComponentType<RootPortalProps> =
  createHostComponent<RootPortalProps>('root-portal');
