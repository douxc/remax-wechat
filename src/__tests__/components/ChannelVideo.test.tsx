import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import { ChannelVideo } from '../../hostComponents';

describe('ChannelVideo', () => {
  it('render correctly', () => {
    const testRenderer = TestRenderer.create(
      <ChannelVideo
        feedId='123'
        finderUserName='test'
        feedToken='token'
        autoplay='true'
      />
    );

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
