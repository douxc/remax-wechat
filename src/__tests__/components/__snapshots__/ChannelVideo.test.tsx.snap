// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelVideo render correctly 1`] = `
<channel-video
  autoplay="true"
  feedId="123"
  feedToken="token"
  finderUserName="test"
  loop={false}
  muted={false}
  objectFit="contain"
/>
`;
